netman/adapters/switches/cached.py

```diff
@@ -0,0 +1,321 @@
+# Copyright 2015 Internap.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import OrderedDict
+
+from netman.core.objects import port_modes
+from netman.core.objects.bond import Bond
+from netman.core.objects.interface import Interface
+from netman.core.objects.switch_base import SwitchBase
+from netman.core.objects.vlan import Vlan
+from netman.core.objects.vrrp_group import VrrpGroup
+
+
+__all__ = ['CachedSwitch']
+
+
+class Cache():
+    object_type = None
+    object_key = None
+    object_extra = {}
+
+    def __init__(self, *args, **kwargs):
```

Comment:
I don't particularly like anonymous params since we need to dig in the code to find out what the params are, and if i do they're all passed to an ordered dict, and reading the calling code an ordered dict takes a list of tuples (key, value), ok.   So i suggest just receving a list of tuples instead of anonymous args